android/src/main/java/io/sentry/capacitor/SentryCapacitor.java

```diff
@@ -142,15 +190,113 @@ public void captureEnvelope(PluginCall call) {
             File installation =  new File(sentryOptions.getOutboxPath(), UUID.randomUUID().toString());
             try (FileOutputStream out = new FileOutputStream(installation)) {
                 out.write(envelope.getBytes(Charset.forName("UTF-8")));
-                logger.info("Successfully captured envelope");
+                logger.info("Successfully captured envelope.");
+
+                JSObject resp = new JSObject();
+                resp.put("value", envelope);
+                call.resolve(resp);
+
             } catch (Exception e) {
-                logger.info("Error writing envelope: ", e);
+                logger.info("Error writing envelope.");
+                call.reject(String.valueOf(e));
```

Comment:
Might want to do a `return` after this just so its clear that nothing should happen outside of this try